tests/watch.test.tsx

```diff
@@ -96,4 +109,56 @@ describe('watch', () => {
 
     reference.value = 'Update'
   })
+  it('should support promise watchers', async () => {
+    const reference = proxy({ value: 'Example' })
+
+    const callback = vi.fn()
+
+    const waitPromise = sleep(10000)
+    watch(async (get) => {
+      await waitPromise
+      get(reference)
+      callback()
+    })
+
+    vi.runAllTimers()
+    await waitPromise
```

Comment:
Do we need this?

Reply:
In this instance yes - because whilst the timers will be run - the code is still synchronous and won't wait for the other scheduled tasks to finish - like the promise itself which is scheduled as a macro task so it would resolve after the current function call. It's similar to why you do `await Promise.resolve()`

Reply:
Oh, yeah. Right. My oversight.